### ISSUE TITLE
UI/UX improvement of badgeyay

### ISSUE THREAD

**Issue by nanspro** [fossasia/badgeyay#210]
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behavior:**
Current UI is normal,it can be made better

**Comment by nanspro** [comment]
As Badgeyay is a small project i don't think there is anything left in it technically so final enhancements which we can do is improve it's UI

**Comment by nanspro** [comment]
@fossasia/badgeyay-admin  What do you guys think about this?

**Comment by meets2tarun** [comment]
@nanspro Great! We would love to see the new UI/UX. Make a PR.

**Comment by gabru-md** [comment]
@nanspro great !

**Comment by meets2tarun** [comment]
@nanspro Please share your progress!

**Comment by vaibhavsingh97** [comment]
@nanspro If you findin any sort of problem than i am happy to collaborate

**Comment by nanspro** [comment]
@meets2tarun  Due to Assignments i was not able to give time but i will work on this today and hopefully submit an PR by the end of the day.

**Comment by nanspro** [comment]
I am thinking about using "Vuejs" for UI improvements.

**Comment by nanspro** [comment]
@fossasia/badgeyay-admin  you can suggest some particular things we can improve in current UI

**Comment by vaibhavsingh97** [comment]
@nanspro Can you use some prototype tool and show us some desings you are thinking of. Then only we can add some inputs 😄 
And Why `Vuejs`? Any specific reason?
We can use Angular v4

**Comment by nanspro** [comment]
@vaibhavsingh97  I am currently  learning Vuejs as this will help me in My major project and i think it's cool

**Comment by jajodiaraghav** [comment]
Hi everyone.
I would like to share my views on this issue:

1. I think integrating SPA packages such as Vue or Angular doesn't make any sense. SPAs are used when we have multiple components in a single frame, all interacting with one another (or external APIs) to make the application seamless. Badgeyay neither interacts with APIs. nor it has multiple components (It has just one : Upload)

2. We can instead integrate a nav-pills frame, which would separate the different functionalities of uploading files for easier navigation.

I would love to know what others think. Feel free to disagree (or correct me if I am wrong).

Hi everyone.
I would like to share my views on this issue:

1. I think integrating SPA packages such as Vue or Angular doesn't make any sense. SPAs are used when we have multiple components in a single frame, all interacting with one another (or external APIs) to make the application seamless. Badgeyay neither interacts with APIs. nor it has multiple components (It has just one : Upload)

2. We can instead integrate a nav-pills frame, which would separate the different functionalities of uploading files for easier navigation.

I would love to know what others think. Feel free to disagree (or correct me if I am wrong).

**Comment by vaibhavsingh97** [comment]
@jajodiaraghav ohh! Agreed with you, If we need UI enhancement we can make use of Bootstrap4 as it's genrally a UI refresh like cards, etc

**Comment by jajodiaraghav** [comment]
@vaibhavsingh97 Why do we need to switch to Bootstrap 4? Is there something that Bootstrap 3 doesn't have that we need?

I was thinking of separating the various upload options in bootstrap tabs. We can also you this (Optional) : https://github.com/kartik-v/bootstrap-tabs-x

**Comment by jajodiaraghav** [comment]
I would like to work on this if nobody is working on this.

**Comment by vaibhavsingh97** [comment]
We haven't finalized, I would like to propose prototype before development
and after that we can go for Development. What are your opinion?

**Comment by ParthS007** [comment]
Yes, It will great @vaibhavsingh97  if we have  some prototype first and then develop which is best from them.

**Comment by jajodiaraghav** [comment]
I was thinking of separating the various upload options in bootstrap tabs. We can also you this (Optional) : https://github.com/kartik-v/bootstrap-tabs-x

**Comment by vaibhavsingh97** [comment]
Hi! i have exams so not able to design prototype but i will make for sure this weekend. If anybody wants to take this, you are free to take

**Comment by nanspro** [comment]
@vaibhavsingh97  i am having similar problems due to my major project and exams i won't be able to do anything till 12th

**Comment by jajodiaraghav** [comment]
I think we should finalize a prototype before proceeding with this one ✊

**Comment by vaibhavsingh97** [comment]
@jajodiaraghav I think me and @nanspro can collborate and submit prototype this saturday.

**Comment by vaibhavsingh97** [comment]
Initial Design
![image](https://user-images.githubusercontent.com/8705386/31623200-3c0a817c-b28e-11e7-8aaf-686933afaa68.png)
Working link: https://pr.to/X3IYGF/
@jajodiaraghav @nanspro @S2606 
Let me know your thoughts on this

**Comment by jajodiaraghav** [comment]
This looks really great! You did a lot of work there. Thanks @vaibhavsingh97 
What are the fields that the "Enter Data" tab would contain? Or should the Enter data part be below the upload CSV section?

I think we should start working on this if others feel the same.

**Comment by vaibhavsingh97** [comment]
@jajodiaraghav I had designed this with reference to Material Design. We have two options either to use [Materializecss](http://materializecss.com/) or [Angular](https://angular.io/). We can't use Angula as we use Angular when we have multiple components in a single frame, all interacting with one another (or external APIs) to make the application seamless. Badgeyay neither interacts with APIs. nor it has multiple components (It has just one : Upload). Only option left is Materializecss. I have choosen materiaizecss over bootstrap as its a front-end framework based on Material Design.
What are your views?

**Comment by S2606** [comment]
Are we considering @djmgit point of decoupling the app for API creation of badgeyay?

**Comment by djmgit** [comment]
@jajodiaraghav @vaibhavsingh97 I too feel materializecss will do good for this work. Although we are not not presently interacting with external APIs, we may still use SPA frameworks like react or angular or vue but it will have a separate overhead, that is server side rendering. As of now since our project is server rendered (coupled), i guess materializecss is a simple and good option :)

**Comment by ParthS007** [comment]
@vaibhavsingh97 @jajodiaraghav  materialize css is good option for this . We can start working on this .

**Comment by jajodiaraghav** [comment]
I think it would be nice to assign a part of task to each and every contributor interested in working on this [Including me :)]. This way all will get a chance to contribute one by one. This should be done in a new branch and should only be merged when completed.

P1 can start with the basic layout, once P1 is done, P2 can integrate the dropdown. This will be followed by P3 doing something else. We should also add both frontend and backend for 'Custom Text'.

Thoughts?

**Comment by vaibhavsingh97** [comment]
@jajodiaraghav And you stole my words. Creating a new branch and working on one part is good idea. 👍

**Comment by ParthS007** [comment]
@jajodiaraghav agree with you :+1:

**Comment by djmgit** [comment]
@jajodiaraghav agreed with you,  nice approach,  would love to collaborate on this :)

**Comment by S2606** [comment]
@jajodiaraghav Ready to work...

**Comment by gabru-md** [comment]
@jajodiaraghav Great! Would love to work on this :)

**Comment by jajodiaraghav** [comment]
Can someone please break it into multiple sub tasks?
I really want to integrate the drag-and-drop upload if others are okay with it.

Thanks :)

**Comment by vaibhavsingh97** [comment]
@jajodiaraghav I will open issues with the subtasks. Also, We need one new branch where we can work on 😄

**Comment by djmgit** [comment]
We also need to implement #284 as suggested bu @mariobehling

**Comment by mariobehling** [comment]
As a matter of fact the required features are not yet there and testing different options I get a bad request. This is not stable yet. Before a next version UI is tackled the app should be more stable. Please postpone.

**Comment by S2606** [comment]
@abishekvashok Since UI is now updated, can we close this??

**Comment by abishekvashok** [comment]
Sure I am closing this as this was meant for an old ui
